### PR TITLE
Resolve typing errors in kernel mocks

### PR DESCRIPTION
**Pull request by ph-ph**
## References
Addresses issue https://github.com/jupyterlab/jupyterlab/issues/8929

## Code changes
I'll comment on specific changes in the diff

## User-facing changes
None

## Backwards-incompatible changes
None that I'm aware of

### CODE REVIEW

File: testutils/src/mock.ts [hunk 1]
```diff
@@ -170,6 +170,10 @@ export const KernelMock = jest.fn<
     ...jest.requireActual('@jupyterlab/services'),
     ...options,
     ...model,
+    model,
+    serverSettings: ServerConnection.makeSettings(
+      options.serverSettings as Partial<ServerConnection.ISettings>
```
Comment (ph-ph) [inline testutils/src/mock.ts:175]: Treating `RecursivePartial` as `Partial` should be fine, I guess?

File: testutils/src/mock.ts [hunk 2]
```diff
@@ -271,6 +275,7 @@ export const SessionConnectionMock = jest.fn<
   const name = kernel?.name || options.model?.kernel?.name || DEFAULT_NAME;
   kernel = kernel || new KernelMock({ model: { name } });
   const model = {
+    id: UUID.uuid4(),
```
Comment (ph-ph) [inline testutils/src/mock.ts:278]: Moved here from line 282. This way, we guarantee that `model` has `id`, but allow `options.model` to override it.

File: testutils/src/mock.ts [hunk 3]
```diff
@@ -388,10 +395,9 @@ export const SessionContextMock = jest.fn<
     path: session.path,
     type: session.type,
     name: session.name,
-    kernel: session.kernel,
```
Comment (ph-ph) [inline testutils/src/mock.ts:391]: No such property in `ISessionContext`

File: testutils/src/mock.ts [hunk 4]
```diff
@@ -826,7 +832,7 @@ namespace Private {
   export function changeKernel(
     kernel: Kernel.IKernelConnection,
     partialModel: Partial<Kernel.IModel>
-  ): Promise<Kernel.IModel> {
+  ): Promise<Kernel.IKernelConnection> {
```
Comment (ph-ph) [inline testutils/src/mock.ts:835]: All happy paths return one of the elements in `RUNNING_KERNELS`, and that array is of type `IKernelConnection`. Fixing return type resolves several type errors above.

Comment (fcollonval) [reply testutils/src/mock.ts:835]: :+1:

File: testutils/src/mock.ts (hunk 2 above)
Comment (fcollonval) [reply testutils/src/mock.ts:278]: :+1:

File: testutils/src/mock.ts (hunk 3 above)
Comment (fcollonval) [reply testutils/src/mock.ts:391]: :+1:

**Review by fcollonval — APPROVED** [review]
LGTM thanks @ph-ph 

> Failing test is unrelated

**Pull request merged**